### PR TITLE
sqlitebrowser: update to 3.11.2

### PR DESCRIPTION
**Pull request by amake**
#### Description

Update to 3.11.2. Unfortunately I can't test on Linux.

I checked `port contents` before and after the update, and everything was the same.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.15.1 19B88
Xcode 11.2.1 11B500 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: databases/sqlitebrowser/Portfile [hunk 1]
```diff
@@ -6,8 +6,8 @@ PortGroup           cmake 1.1
 PortGroup           github 1.0
 PortGroup           app 1.0
 
-github.setup        sqlitebrowser sqlitebrowser 3.9.1 v
-revision            1
+github.setup        sqlitebrowser sqlitebrowser 3.11.2 v
+revision            0
```
Comment (RJVB) [inline databases/sqlitebrowser/Portfile]: we don't usually keep 0 revisions

File: databases/sqlitebrowser/Portfile [hunk 2]
```diff
@@ -56,9 +58,8 @@ if {[info procs cmake.save_configure_cmd] ne ""} {
 }
 
 configure.args-append \
-                    -DUSE_QT5=ON \
                     -DQSCINTILLA_INCLUDE_DIR=${qt_includes_dir} \
-                    -DQSCINTILLA_LIB_DIR=${qt_frameworks_dir}
+                    -DQT_LIBRARY_DIR=${qt_frameworks_dir}
```
Comment (RJVB) [inline databases/sqlitebrowser/Portfile]: You'd have to do the same for the Linux hunk.

File: databases/sqlitebrowser/Portfile (hunk 1 above)
Comment (amake) [reply databases/sqlitebrowser/Portfile]: There are 1,677 ports with a 0 revision, but if you feel strongly about it I will remove it.

File: databases/sqlitebrowser/Portfile (hunk 2 above)
Comment (amake) [reply databases/sqlitebrowser/Portfile]: Thanks. Done.

**Comment by amake** [comment]
> As you may have seen I have an additional patch in my personal version of the port

I didn't know you had a personal version. Where can I see it?

> have you simply not run into the warning popups?

I haven't seen that, but I basically only use this app to inspect local DB files. I assume those popups are only related to the "Remote" feature? Or could the be related to version checking? I noted that version checking was not disabled in the previous port version; I have disabled it this time.

**Comment by RJVB** [comment]
>I didn't know you had a personal version. Where can I see it?

https://github.com/RJVB/macstrop/tree/master/databases/sqlitebrowser

>I haven't seen that, but I basically only use this app to inspect local DB files. I assume those popups are only related to the "Remote" feature? Or could the be related to version checking? I noted that version checking was not disabled in the previous port version; I have disabled it this time.

No, not version checking. Annoyingly I don't remember the details of how I got the warning popups but it's got to do with remote database access. The patch isn't crucial but not patching the code effectively means that the remote feature cannot work (because there's no such thing as a network manager on Mac, AFAIK).

Re: the revision: I've been told in the past to remove the entire line if the revision is 0, and it certainly does make it a bit easier not to forget to reset the revision after a version upgrade. I don't really care either way.

File: databases/sqlitebrowser/Portfile (hunk 1 above)
Comment (mf2k) [reply databases/sqlitebrowser/Portfile]: Indeed, yes we do now.

**Comment by amake** [comment]
> The patch isn't crucial but not patching the code effectively means that the remote feature cannot work (because there's no such thing as a network manager on Mac, AFAIK).

I just tried the remote feature (made an account on dbhub.io, exported my certificate, opened another user's public DB) and it worked fine for me.

**Comment by RJVB** [comment]
>I just tried the remote feature (made an account on dbhub.io, exported my certificate, opened another user's public DB) and it worked fine for me.

What Qt version? If memory serves me well the network manager feature (with things like the bearer plugin) has been overhauled since the Qt version I'm using (5.9).

Do you know if the remote feature exists in the version currently in MacPorts? If it does we can leave you the patch, if it doesn't I'd prefer to put it in just to avoid having to do it retro-actively if people start complaining (because that will land on my plate and I'd have to find someone to commit the patch for me).

Of course, if you have commit access you're welcome to add yourself as a co-maintainer. Or even *the* maintainer given you actually seem to use the application more than once or twice a year like I do ;)

**Comment by amake** [comment]
In v3.9.1 the Remote feature was merely a placeholder that opened https://dbhub.io: https://github.com/sqlitebrowser/sqlitebrowser/commit/56107c958d145e58143c900efd554e0e8b14d2d4

The Qt version that was chosen automatically by the qt5-1.0 PortGroup on my machine is 5.13.2.

Since the remote feature works fine on at least some Qt versions, I wouldn't want to add the patch unconditionally. Do we know what versions are problematic?

**Comment by RJVB** [comment]
On Thursday November 14 2019 04:51:36 Aaron Madlon-Kay wrote:

>Since the remote feature works fine on at least some Qt versions, I wouldn't want to add the patch unconditionally. Do we know what versions are problematic?

No. I can test again with Qt 5.9 if you have a recipe that works for you and that I can follow. But again, there is no such thing as a NetworkManager on Mac; on Linux this is responsible for controlling (and letting the user control) how an internet connection is made. Very few applications should be concerned that kind of problem, and checking the availability of a network manager to determine whether or not there's an internet connection isn't an overly appropriate test .

If I were to guess I'd say that the changes to the relevant Qt components must have been backported to 5.12LTS, so the patch could include a version-check (or be applied conditionally).

**Comment by amake** [comment]
I just tested with Qt 5.9  (I deactivated all my existing Qt 5 ports, modified the qt5 PortGroup to provide `qt59` for my OS, and rebuilt sqlitebrowser) and got no warning popups when using the remote feature.

> But again, there is no such thing as a NetworkManager on Mac

I don't understand this point. I don't know basically anything about Qt, but `QNetworkAccessManager` appears to be a Qt API providing information about network connections. It doesn't seem to be related to any OS-level feature that macOS might or might not have; I would expect that under the hood it uses whatever is appropriate for the OS.

Reference: [QNetworkAccessManager::networkAccessible()](https://code.qt.io/cgit/qt/qtbase.git/tree/src/network/access/qnetworkaccessmanager.cpp?h=v5.13.0#n1113)

**Comment by kencu** [comment]
This looks ready to commit -- is that right?

**Comment by RJVB** [comment]
Certainly not wrong AFAIC :)

**Pull request merged**